model/stacktrace.go

```diff
@@ -7,12 +7,52 @@ import (
 
 type Stacktrace []*StacktraceFrame
 
-func (st *Stacktrace) Transform(config *pr.Config, service Service) []common.MapStr {
-	var frames []common.MapStr
+func (st Stacktrace) Transform(config *pr.Config, service Service) []common.MapStr {
+	if config == nil || config.SmapMapper == nil {
+		return st.transform(config)
+	}
+	return st.sourcemapAndTransform(config, service)
+}
 
-	for _, fr := range *st {
-		frame := fr.Transform(config, service)
+func (st Stacktrace) transform(config *pr.Config) []common.MapStr {
+	var frames []common.MapStr
+	for _, fr := range st {
+		frame := fr.Transform(config)
 		frames = append(frames, frame)
 	}
 	return frames
 }
+
+// source map algorithm:
+// apply source mapping frame by frame
+// if no source map could be found, set updated to false and set sourcemap error
+// otherwise use source map library for mapping and update
+// - filename: only if it was found
+// - function:
+//   * should be moved down one stack trace frame,
+//   * the function name of the first frame is set to <anonymous>
```

Comment:
it's only set to `<anonymous>` if there's a problem, AFAICT.

Reply:
Not exactly, `<anonymous>` is set for the first frame. As the logic is to always use the sourcemapped function name of the previous frame, the first frame doesn't have a predecessor so it gets an <anonymous>.
For all the others, if no function name can be found for a frame,  `<unknown>` is set as function name for the next frame.